src/Umbraco.Core/Xml/XmlHelper.cs

```diff
@@ -229,8 +228,9 @@ public static XmlDocument OpenAsXmlDocument(string filePath)
         /// <returns>a XmlAttribute</returns>
         public static XmlAttribute AddAttribute(XmlDocument xd, string name, string value)
         {
-            if (xd == null) throw new ArgumentNullException("xd");
-            if (string.IsNullOrEmpty(name)) throw new ArgumentNullOrEmptyException(nameof(name));
+            if (xd == null) throw new ArgumentNullException(nameof(xd));
+            if (name == null) throw new ArgumentNullException(nameof(name));
+            if (string.IsNullOrEmpty(name)) throw new ArgumentException("Value can't be empty.", nameof(name));
```

Comment:
`SetAttribute` uses `string.IsNullOrWhiteSpace()`: should we change this to keep it the same?

Reply:
Yep IsNullOrWhiteSpace is better